### PR TITLE
LibGfx: Use StringView literals for CSS color list

### PR DESCRIPTION
**Pull request by MacDue**
And also remove the null terminating entry in favour of for-each loop.

(Randomly stumbled across this and thought it could use a minor spruce up, this was 2019 code :^))

### CODE REVIEW

File: Userland/Libraries/LibGfx/Color.cpp [hunk 1]
```diff
@@ -83,178 +83,171 @@ Optional<Color> Color::from_string(StringView string)
...
-        constexpr ColorAndWebName(ARGB32 c, char const* n)
-            : color(c)
-            , name(n != nullptr ? StringView { n, __builtin_strlen(n) } : StringView {})
-        {
-        }
         ARGB32 color;
         StringView name;
     };
 
     constexpr ColorAndWebName web_colors[] = {
```
Comment (linusg) [inline Userland/Libraries/LibGfx/Color.cpp:90]: make that an `AK::Array` while we're here, pls :)

File: Userland/Libraries/LibGfx/Color.cpp [hunk 2]
```diff
@@ -83,178 +83,171 @@ Optional<Color> Color::from_string(StringView string)
...
+        { 0x663399, "rebeccapurple"sv },
     };
 
     if (string.equals_ignoring_ascii_case("transparent"sv))
         return Color::from_argb(0x00000000);
 
-    for (size_t i = 0; !web_colors[i].name.is_null(); ++i) {
-        if (string.equals_ignoring_ascii_case(web_colors[i].name))
-            return Color::from_rgb(web_colors[i].color);
+    for (auto& web_color : web_colors) {
```
Comment (linusg) [inline Userland/Libraries/LibGfx/Color.cpp:248]: `auto const&`

File: Userland/Libraries/LibGfx/Color.cpp (hunk 1 above)
Comment (MacDue) [reply Userland/Libraries/LibGfx/Color.cpp:90]: Then we have to say `ColorAndWebName` for each entry or it won't infer the length (which is super annoying)

Comment (MacDue) [reply Userland/Libraries/LibGfx/Color.cpp]: I guess it's not too bad with a nicer name :)

File: Userland/Libraries/LibGfx/Color.cpp [hunk 3]
```diff
@@ -82,179 +82,172 @@ Optional<Color> Color::from_string(StringView string)
...
-            , name(n != nullptr ? StringView { n, __builtin_strlen(n) } : StringView {})
-        {
-        }
+    struct WebColor {
         ARGB32 color;
         StringView name;
     };
 
-    constexpr ColorAndWebName web_colors[] = {
+    Array web_colors {
```
Comment (linusg) [inline Userland/Libraries/LibGfx/Color.cpp:90]: Still can be `constexpr`, no?

File: Userland/Libraries/LibGfx/Color.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibGfx/Color.cpp]: Yeah, seems ok. I'm sure some template magician can figure out how to make it infer the type and length eventually!

**Pull request merged**